joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.h

```diff
@@ -246,6 +252,54 @@ class JointTrajectoryController : public controller_interface::Controller<Hardwa
    */
   void setHoldPosition(const ros::Time& time, RealtimeGoalHandlePtr gh=RealtimeGoalHandlePtr());
 
+protected:
+  /**
+   * @returns the number of joints of the robot.
+   */
+  unsigned int getNumberOfJoints() const;
+
+  /**
+   * @brief Updates the states by sampling the specified trajectory for each joint
+   * at the specified sampling time.
+   *
+   * The current state is updated based on the values transmitted by the
+   * corresponding JointHandles.
+   *
+   * @param sample_time Time point at which the joint trajectories have to be sampled.
+   * @param traj Trajectory containing all joint trajectories currently under execution.
+   *
+   * @note This function is NOT thread safe but intended to be used in the
+   * update-function.
+   */
+  void updateStates(const ros::Time& sample_time, const Trajectory* const traj);
+
+protected:
+  /**
+   * @returns a trajectory consisting of joint trajectories with one pre-allocated
+   * segment.
+   */
+  static TrajectoryPtr createHoldTrajectory(const unsigned int& number_of_joints);
```

Comment:
For the Doxygen Html output it would be better to state what the function does via `@brief` statement instead of using the `@returns` statement. If `@brief` is used, Doxygen displays the explanation what the function does already in the summary at the beginning of the Html page. What do you think?

Reply:
Done.